packages/sales-report-tool/index.js

```diff
@@ -0,0 +1,107 @@
+const fastcsv = require("fast-csv");
+const fs = require("fs");
+const ws = fs.createWriteStream("data.csv");
+const fetch = require("cross-fetch");
+
+// Data formating
+const getFormattedDate = (date) => {
+  const theDate = new Date(date * 1000);
+  const dateString = theDate.toGMTString();
+  return dateString;
+};
+
+// Array for storing transactions
+const arrOfTxn = [];
+let total;
+// Minting wallet address
+const address = `SP2X0TZ59D5SZ8ACQ6YMCHHNR2ZN51Z32E2CJ173`;
+// Mint price
+const mintPrice = 100;
+
+const getAccounting = async () => {
+  await fetch(
+    `https://stacks-node-api.mainnet.stacks.co/extended/v1/address/${address}/transactions`
+  )
+    .then((response) => response.json())
+    .then((data) => {
+      total = data.total;
+    });
+  let indexOfLastTxn = total;
+  let index = 1;
+  let nftsSold = 0;
+
+  while (indexOfLastTxn > 0 && indexOfLastTxn <= total) {
+    await fetch(
+      `https://stacks-node-api.mainnet.stacks.co/extended/v1/address/${address}/transactions?offset=${
+        indexOfLastTxn - 20
+      }`
+    )
+      .then((response) => response.json())
+      .then((data) => {
+        data.results &&
+          data.results.length > 0 &&
+          data.results.map((txn) => {
+            if (
+              txn.sender_address !== address &&
+              txn.tx_type === "contract_call" &&
+              txn.post_conditions[0]
+            ) {
+              nftsSold =
+                nftsSold + txn.post_conditions[0].amount / 1000000 / mintPrice;
+              arrOfTxn.push({
+                number: index,
+                time: getFormattedDate(txn.burn_block_time),
+                txnId: txn.tx_id,
+                address: txn.sender_address,
+                amountOfSTX: txn.post_conditions[0].amount / 1000000,
```

Comment:
relying on the postcondition might not be the best idea there, would be better to use the function name and map it to the correct number.
Could also mean that the function can be simplified and you can handle both the contract and cli calls at the same time if I get it properly